cdk/apps/slack/stacks/slack_lambda.py

```diff
@@ -77,17 +77,30 @@ def __init__(self, scope: core.Construct, id: str, slack_channel: str, **kwargs)
             role=lambda_role
         )
 
-        rule = _events.Rule(
-            self,
-            'BatchEventToSlackLambda'
-        )
-        rule.add_event_pattern(
-            source=['aws.batch'],
-            detail_type=['Batch Job State Change'],
-            detail={'status': [
-                'FAILED',
-                'SUCCEEDED',
-                'RUNNABLE'
-            ]}
-        )
-        rule.add_target(_events_targets.LambdaFunction(handler=function))
+        batch_job_queues = {
+            'AGHA':           {'name': 'agha-job-queue',                         'enabled': True},
+            'Umccrise':       {'name': 'cdk-umccrise_job_queue',                 'enabled': True},
+            'UmccriseDragen': {'name': 'cdk-umccrise_job_queue-dragen-testing',  'enabled': False},
+            'Nextflow':       {'name': 'nextflow-job-queue',                     'enabled': False},
+            'WtsReport':      {'name': 'wts_report_batch_queue_dev',             'enabled': True},
+        }
+        for job_queue_id, job_queue_config in batch_job_queues.items():
+            rule = _events.Rule(
+                self,
+                f'BatchEventToSlackLambda{job_queue_id}',
+                enabled=job_queue_config['enabled'],
+                event_pattern=_events.EventPattern(
+                    detail={
+                        'status': [
+                            'FAILED',
+                            'SUCCEEDED',
+                            'RUNNABLE',
+                        ],
+                        'jobQueue': job_queue_config['name'],
```

Comment:
Thinking further about this, the value of `jobQueue` will likely need to be set as the job queue ARN rather than the job queue name. The CDK JobQueue class has a [method](https://docs.aws.amazon.com/cdk/api/latest/python/aws_cdk.aws_batch/JobQueue.html#aws_cdk.aws_batch.JobQueue.from_job_queue_arn) to look up a defined job queue but the lookup itself requires the ARN.

Is it safe to construct ARNs where the resource id is known as shown [here](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html#arns-syntax)? If so, this could be alternative.

Reply:
Yes, I think that has to be an ARN and yes, it should be safe to construct the ARN. The only variable parts are the account ID (which you can get from CDK) and the queue name (which you already have there).